cmd/api/routers/routers.go

```diff
@@ -41,43 +41,33 @@ type List struct {
 	Metadata   map[string]string            `json:"metadata"`
 }
 
-// GetAllResources responds with the list of resources of a specific type across all namespaces
-func (c *Controller) GetAllResources(context *gin.Context) {
-	group := context.Param("group")
-	version := context.Param("version")
-	apiVersion := fmt.Sprintf("%s/%s", group, version)
+func (c *Controller) GetResources(context *gin.Context) {
+	var err error
 	limit, id, date := pagination.GetValuesFromContext(context)
-
 	kind, err := discovery.GetAPIResourceKind(context)
 	if err != nil {
 		abort.Abort(context, err, http.StatusInternalServerError)
 		return
 	}
 
-	resources, lastId, lastDate, err := c.Database.QueryResources(context.Request.Context(), kind, apiVersion, limit, id, date)
-	if err != nil {
-		abort.Abort(context, err, http.StatusInternalServerError)
-		return
-	}
-
-	continueToken := pagination.CreateToken(lastId, lastDate)
-	context.JSON(http.StatusOK, NewList(resources, continueToken))
-}
-
-func (c *Controller) GetNamespacedResources(context *gin.Context) {
 	group := context.Param("group")
 	version := context.Param("version")
-	apiVersion := fmt.Sprintf("%s/%s", group, version)
 	namespace := context.Param("namespace")
-	limit, id, date := pagination.GetValuesFromContext(context)
 
-	kind, err := discovery.GetAPIResourceKind(context)
-	if err != nil {
-		abort.Abort(context, err, http.StatusInternalServerError)
-		return
+	apiVersion := version
+	if group != "" {
+		apiVersion = fmt.Sprintf("%s/%s", group, version)
+	}
+
+	var resources []*unstructured.Unstructured
+	var lastId int64
+	var lastDate string
+	if namespace != "" {
+		resources, lastId, lastDate, err = c.Database.QueryNamespacedResources(context.Request.Context(), kind, apiVersion, namespace, limit, id, date)
+	} else {
+		resources, lastId, lastDate, err = c.Database.QueryResources(context.Request.Context(), kind, apiVersion, limit, id, date)
 	}
```

Comment:
I believe that we could make a similar merge of the Database `QueryNamespacedResources` and `QueryResources`, so we could replace this if/else block to a one call. 

I am ok doing that in a separate PR though.

Reply:
I gave it a try, let me know if it looks good.